First code review

tomasesquivelgc opened the pull request:
## Development branch includes:

 - Items fetched from TV shows API are JS generated in the document.
 - Each show card includes a like button and a like counter, a comment button, and a reserve button.
 - comment and reserve buttons both generate popUp cards containing show information, API fetch elements, and an input area to submit your own comment/reservation.
 - comments and reservation information are also fetched from a different API (involvement API)
 - You can also find an item counter on the header.
 - Readme file with detailed information on how to run the project, a project description including a demonstration video with the team behind the project, and a live demo.
 - Jest tests for all counter functions (commentCounter, likeCounter and reserveCounter).

hafiz1379 committed:
Added babelrc

hafiz1379 committed:
Added jest.config.js

hafiz1379 committed:
Updated

hafiz1379 committed:
Added script

hafiz1379 committed:
Updated

hafiz1379 committed:
Updated

hafiz1379 committed:
Deleted

hafiz1379 committed:
Deleted secer.js

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Merge pull request #21 from hafiz1379/Add-jest-testing-library

Add jest testing library

tomasesquivelgc committed:
add colors as variables

tomasesquivelgc committed:
Merge pull request #23 from hafiz1379/choosing-assets

add colors as variables

tomasesquivelgc committed:
Revert "add colors as variables"

tomasesquivelgc committed:
Merge pull request #25 from hafiz1379/revert-23-choosing-assets

Revert "add colors as variables"

hafiz1379 committed:
Added logo.png into pro

hafiz1379 committed:
Added logo and nav into header

hafiz1379 committed:
Added style for logo and nav

hafiz1379 committed:
Import style.css

hafiz1379 committed:
Fixed linter error

mdg0410 committed:
Add footer section | Add style at footer section

hafiz1379 committed:
Fixed the issue of loading logo

hafiz1379 committed:
Merge pull request #26 from hafiz1379/Add-header

Add header

mdg0410 committed:
Merge branch 'Development' into addFooter

mdg0410 committed:
Resolve conflicts | Add copyright element

mdg0410 committed:
Change tag <section> for <footer>

mdg0410 committed:
Merge pull request #28 from hafiz1379/addFooter

Add footer section

hafiz1379 committed:
Added font

hafiz1379 committed:
Added font

hafiz1379 committed:
Added font

hafiz1379 committed:
Added dic for movie cards

hafiz1379 committed:
Added API url and functions

hafiz1379 committed:
Updated

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Fixed linter errors

hafiz1379 committed:
Merge pull request #29 from hafiz1379/Display-all-items

Display all items

tomasesquivelgc committed:
add showComments module

add id as an property of movieData
add eventlistener to every comment button.
function takes the id of the show and console logs it.

tomasesquivelgc committed:
comment card create

function that creates comment card create

tomasesquivelgc committed:
add input fields to comment card

tomasesquivelgc committed:
add close button to popCard

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
Merge pull request #30 from hafiz1379/comment-pop-up

Comment pop up

tomasesquivelgc committed:
fetchData function create

function logs data from API when the comment button is clicked

mdg0410 committed:
Add pop up for reserve film

mdg0410 committed:
Add function close pop up for reserve film

mdg0410 committed:
Resolve problem with display pop up reserve film

mdg0410 committed:
Add new style in pop up reserve film

mdg0410 committed:
resize image in pop-up window to reserve movie

mdg0410 committed:
Code cleaning and restructuring

mdg0410 committed:
Fix & Solver errors

mdg0410 committed:
Fix & Solver errors

mdg0410 committed:
Fix & Solver errors

mdg0410 committed:
Change word `Gender` to `Genders`

mdg0410 committed:
Merge pull request #31 from hafiz1379/reserve-pop-up

Reserve pop up

tomasesquivelgc committed:
fetchData function finish

function now displays the comments in the commentSection when the popUp is constructed.
comments add to the first two element as examples

tomasesquivelgc committed:
Merge branch 'Development' into comment-pop-up

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
Merge branch 'comment-pop-up' of https://github.com/hafiz1379/JS-capstone-project into comment-API-fetch

hafiz1379 committed:
Added display number of likes capability

hafiz1379 committed:
Added likesApiURL

hafiz1379 committed:
Added style for number of likes

hafiz1379 committed:
updated

hafiz1379 committed:
updated

hafiz1379 committed:
Fixed linter errors

mdg0410 committed:
Add loadSaveReserve file | Add function async for reserve film

hafiz1379 committed:
Fixed theissue

hafiz1379 committed:
Merge pull request #33 from hafiz1379/Displau-number-of-likes

Display number of likes

tomasesquivelgc committed:
run build

tomasesquivelgc committed:
Merge branch 'Development' into comment-pop-up

tomasesquivelgc committed:
Merge pull request #32 from hafiz1379/comment-pop-up

Comment fetch from API

mdg0410 committed:
Add style in the reserve list section | Add function to close `popup reserve` with one click on main page

mdg0410 committed:
Merge branch 'Development' into reserve-pop-up

mdg0410 committed:
Fix & Solver errors

mdg0410 committed:
Fix & Solved errors

mdg0410 committed:
Fix & Solved errors

mdg0410 committed:
Merge pull request #34 from hafiz1379/reserve-pop-up

Reserve fetch from API

hafiz1379 committed:
Added new like feature

hafiz1379 committed:
Added new like feature

hafiz1379 committed:
Merge pull request #35 from hafiz1379/Add-new-like

Add new like

tomasesquivelgc committed:
savedata function create

savedata in charge of saving comments to the API

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
Merge pull request #36 from hafiz1379/saveComments

Save comments

tomasesquivelgc committed:
comment counter add

tomasesquivelgc committed:
manage empty inputs

tomasesquivelgc committed:
placeholder text changes for empty messages

tomasesquivelgc committed:
linters fix

hafiz1379 committed:
Added movie counter fun

hafiz1379 committed:
Imported movieCount.js

hafiz1379 committed:
Added id for movie option

hafiz1379 committed:
Added id for movie option

hafiz1379 committed:
Updated

hafiz1379 committed:
Resolved merge conflict by keeping README.md file.

hafiz1379 committed:
Fixed the issue

hafiz1379 committed:
Fixed linters

hafiz1379 committed:
Fixed

hafiz1379 committed:
Fixed

hafiz1379 committed:
Fixed

tomasesquivelgc committed:
Merge branch 'Development' into Add-items-counter

hafiz1379 committed:
Fixed lintert error

hafiz1379 committed:
Merge branch 'Add-items-counter' of github.com:hafiz1379/JS-capstone-project into Add-items-counter

hafiz1379 committed:
Merge pull request #38 from hafiz1379/Add-items-counter

Add items counter

tomasesquivelgc committed:
Merge branch 'Development' into comment-counter

tomasesquivelgc committed:
pull origin run build

final check before merge

tomasesquivelgc committed:
Merge pull request #37 from hafiz1379/comment-counter

Comment counter

mdg0410 committed:
Add form to book a movie | Add function for save reservation

mdg0410 committed:
Add function for count the number of reservations

mdg0410 committed:
Create enviroment for work with `Jest`and `Mocks` | Add `counterReserve.test.js`, `counterReserveMock.js` and `reservationsMock.js` files |

mdg0410 committed:
Fix & Solved errors

mdg0410 committed:
.gitignore Fixed

mdg0410 committed:
Merge pull request #40 from hafiz1379/saveReserve

Save reservations

mdg0410 committed:
Merge pull request #41 from hafiz1379/counterReserve

Reservation counter

mdg0410 committed:
Add descriptions to the `counting reserves` tests.

mdg0410 committed:
Merge pull request #42 from hafiz1379/testCounterReserve

Testing reserve counter

hafiz1379 committed:
Added movieCount.test.js

hafiz1379 committed:
Added movieCount.test.js

hafiz1379 committed:
Updated

hafiz1379 committed:
Fixed the linters

hafiz1379 committed:
Merge pull request #45 from hafiz1379/Items-test

Items test

tomasesquivelgc committed:
fix google dependency

to avoid unnescessary node_modules that conflict with node.js

tomasesquivelgc committed:
files created

first tests add

tomasesquivelgc committed:
tests add, all pass

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
Merge pull request #47 from hafiz1379/comment-counter-test

Comment counter test

mdg0410 committed:
Update README file

mdg0410 committed:
Update README file

mdg0410 committed:
Merge pull request #49 from hafiz1379/updateReadmeFile

Update README file

tomasesquivelgc committed:
video add to project description

project demo link fix

tomasesquivelgc committed:
Merge pull request #50 from hafiz1379/adding-vide-to-README

video add to project description

ishimwezachee commented on src/index.html:
```diff
@@ -1,11 +1,29 @@
+        <li><a href="">Contact</a></li>
+      </ul>
+    </nav>
+    <div id="movie-cards"></div>
+    <div id="reserveCard"></div>
```
- [ ]  You comments, likes and reservation are functioning properly , but they look different from the provided mockup.  Make sure to use the icons for likes, and buttons for comments and Reservations. it will make your application more beautiful and professional. 
![image](https://github.com/hafiz1379/JS-capstone-project/assets/43877689/5464c78c-6462-4407-b796-0ed7f77abc07)

- [ ]  You should write Reserve as a full word, not cutting it for the best user experience.  Just write reserve in full word.

tomasesquivelgc replied:
Hello @ishimwezachee , would you mind if I ask for some clarification? The comment and reserve buttons do not present any icons on the mockup. I understand we should change the like button for a different icon but what did you mean by the comment and reserve buttons? to be clear, those 2 are in fact buttons, so I don't understand your correction.

Whoistolu replied:
- Well resolved 👍🏽 
- [ ] [OPTIONAL] You could make it `Reservations` and not `reservations` to enhance more professionalism 👍🏽

ishimwezachee commented on src/index.html:
```diff
@@ -1,11 +1,29 @@
+    <title>TV Show</title>
+  </head>
+  <body id="body">
+    <nav>
+        <div>
+          <img class="logo" src="./images/nitro-logo.png" alt="logo">
+        </div>
+      <ul>
+        <li> <a href="">Movies <span id="span">(0)</span></a></li>
+        <li><a href="">Genres</a></li>
+        <li><a href="">Contact</a></li>
+      </ul>
+    </nav>
```
- [ ]  Your navigation header is a little bit different from the provided mockup, please make sure that there is no such space provided as it is designed from the mockup . 
![image](https://github.com/hafiz1379/JS-capstone-project/assets/43877689/df62f5ed-5e9c-4a3e-8046-5b15e23c6698)

tomasesquivelgc replied:
@ishimwezachee  Would this be okay? 
![image](https://github.com/hafiz1379/JS-capstone-project/assets/117373513/0c1ddd9c-7a83-4f48-9345-1550f1f9b458)
The elements were shifted to the flex start to resemble the mockup.

ishimwezachee commented on src/index.html:
```diff
@@ -1,11 +1,29 @@
+    </nav>
+    <div id="movie-cards"></div>
+    <div id="reserveCard"></div>
+    <footer>
+      <div id="footer">
+        <p id="textFooter">Created by Nitro Shows under CC license</p>
+        <p id="copyright">Copyright Nitro Shows<br>2020-2023<br>in collaboration with<br>Microverse</p>
+      </div>
+    </footer>
```
- [ ]  Make sure that the provided footer is similar to the one provided from the design.  seemed like you've added something we don't have from the mockup, while it has to resemble it. 
![image](https://github.com/hafiz1379/JS-capstone-project/assets/43877689/4ab41a69-225a-43ec-af3f-d1098d67637d)

Whoistolu replied:
- Well resolved team 👍🏽

ishimwezachee commented on src/modules/__test__/commentCounter.test.js:
```diff
@@ -0,0 +1,26 @@
+commentNumber.classList.add('commentNumber');
+document.body.appendChild(commentNumber);
+
+// Positive test case
+test('commentCounter should update comment title with correct number of comments', () => {
+  const data = ['comment1', 'comment2', 'comment3'];
+  commentCounter(data);
+  expect(document.querySelector('.commentNumber').innerHTML).toBe(`Comments (${data.length})`);
+});
+
+// Negative test case
+test('commentCounter should update comment title if data is empty', () => {
+  const data = [];
+  commentCounter(data);
+  expect(document.querySelector('.commentNumber').innerHTML).toBe(`Comments (${data.length})`);
+});
+
+// Edge test case
+test('commentCounter should handle large data sets', () => {
+  const data = new Array(1000000).fill('comment');
+  commentCounter(data);
+  expect(document.querySelector('.commentNumber').innerHTML).toBe(`Comments (${data.length})`);
```
- [ ]  Like you did in movies counter, You should cover all edges cases, think of the scenario in which there 0 items on the page ( where counter return zero)

tomasesquivelgc replied:
Hello @ishimwezachee   , me again. I was in charge of comments and it's testing. My second test for this function takes into account the possibility of data being empty, as you can see on my second test: 'commentCounter should update comment title if data is empty'. when that happens the comment counter updates to 0 as opposed to hardcoding 0 as a starting point. Would you mind expanding on your point or correcting me please? otherwise I don't really know what I'm supposed to do. thank you.

Whoistolu replied:
- Well resolved 👍🏽

ishimwezachee commented on src/modules/__test__/counterReserve.test.js:
```diff
@@ -0,0 +1,24 @@
+/* @jest-environment jsdom */
+
+import {
+  film1, film2, film3, film4,
+} from '../__Mocks__/reservationsMock.js';
+import counterReserveJest from '../__Mocks__/counterReserveMock.js';
+
+describe('Testing reservations counter ', () => {
+  test('The first test movie has four reservations, the function is expected to return the numerical value 4.', () => {
+    expect(counterReserveJest(film1)).toBe(4);
+  });
+
+  test('The second test movie has five reservations, the function is expected to return the numerical value 5.', () => {
+    expect(counterReserveJest(film2)).toBe(5);
+  });
+
+  test('The third test movie has one reservations, the function is expected to return the numerical value 1.', () => {
+    expect(counterReserveJest(film3)).toBe(1);
+  });
+
+  test('The fourth test movie has three reservations, the function is expected to return the numerical value 3.', () => {
+    expect(counterReserveJest(film4)).toBe(3);
+  });
+});
```
- [ ]  Like you did in movies counter, You should cover all edges cases, think of the scenario in which there 0 items on the page ( where counter return zero)

Whoistolu replied:
- [ ] Well resolved 👍🏽

ishimwezachee reviewed (changes requested):
Hi @tomasesquivelgc  and @hafiz1379  and @mdg0410 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
![image](https://www.ipem-market.com/wp-content/uploads/Image1.gif)

##  Highlights 
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

tomasesquivelgc committed:
header style fix

tomasesquivelgc committed:
footer fix

tomasesquivelgc committed:
test 0 add, test pass

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
reservations buttons wording change

tomasesquivelgc committed:
Merge pull request #52 from hafiz1379/correcting-styling

Correcting styling

tomasesquivelgc committed:
Merge pull request #53 from hafiz1379/add-test-case-0-to-counterReserve

test 0 add, test pass

hafiz1379 committed:
Added font awesome link

hafiz1379 committed:
Added font awesome link

hafiz1379 committed:
Added like icon inot like btn

hafiz1379 committed:
Added padding for like icon

hafiz1379 committed:
Upadate

hafiz1379 committed:
Fixed linter error

hafiz1379 committed:
Fixed linters

tomasesquivelgc committed:
commentcounter takes 0 into account

note: this change does nothing. It's only here to appease code reviewers. this case was already contemplated BUT some people apparently can't pay attention

tomasesquivelgc committed:
linters fix

tomasesquivelgc committed:
Merge pull request #54 from hafiz1379/comment-counter-test-0

commentcounter takes 0 into account

Whoistolu reviewed (approved):
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
